server/sonar-server/src/main/java/org/sonar/server/rule/CommonRuleDefinitionsImpl.java

```diff
@@ -0,0 +1,139 @@
+      .setDebtRemediationFunction(rule.debtRemediationFunctions().linear("5min"))
+      .setEffortToFixDescription("number of uncovered conditions")
+      .setSeverity(Severity.MAJOR);
+    rule.createParam("minimumBranchCoverageRatio")
+      .setName("The minimum required branch coverage ratio")
+      .setDefaultValue("65")
+      .setType(RuleParamType.FLOAT);
+  }
+
+  private void defineLineCoverageRule(RulesDefinition.NewRepository repo) {
+    RulesDefinition.NewRule rule = repo.createRule("InsufficientLineCoverage");
+    rule.setName("Lines should have sufficient coverage by unit tests")
+      .addTags("bad-practice")
+      .setHtmlDescription("An issue is created on a file as soon as the line coverage on this file is less than the required threshold. " +
+        "It gives the number of lines to be covered in order to reach the required threshold.")
+      .setDebtSubCharacteristic(RulesDefinition.SubCharacteristics.UNIT_TESTABILITY)
+      .setDebtRemediationFunction(rule.debtRemediationFunctions().linear("2min"))
+      .setEffortToFixDescription("number of lines under the coverage threshold")
+      .setSeverity(Severity.MAJOR);
+    rule.createParam("minimumLineCoverageRatio")
+      .setName("The minimum required line coverage ratio")
+      .setDefaultValue("65")
+      .setType(RuleParamType.FLOAT);
+  }
+
+  private void defineCommentDensityRule(RulesDefinition.NewRepository repo) {
+    RulesDefinition.NewRule rule = repo.createRule("InsufficientCommentDensity");
+    rule.setName("Source files should have a sufficient density of comment lines")
+      .addTags("convention")
+      .setHtmlDescription("An issue is created on a file as soon as the density of comment lines on this file is less than the required threshold. " +
+        "The number of comment lines to be written in order to reach the required threshold is provided by each issue message.")
+      .setDebtSubCharacteristic(RulesDefinition.SubCharacteristics.UNDERSTANDABILITY)
+      .setDebtRemediationFunction(rule.debtRemediationFunctions().linear("2min"))
+      .setEffortToFixDescription("number of lines required to meet minimum density")
+      .setSeverity(Severity.MAJOR);
+    rule.createParam("minimumCommentDensity")
+      .setName("The minimum required comment density")
+      .setDefaultValue("25")
+      .setType(RuleParamType.FLOAT);
+  }
+
+  private void defineDuplicatedBlocksRule(RulesDefinition.NewRepository repo) {
+    RulesDefinition.NewRule rule = repo.createRule("DuplicatedBlocks");
+    rule.setName("Source files should not have any duplicated blocks")
+      .addTags("pitfall")
+      .setHtmlDescription("An issue is created on a file as soon as there is at least one block of duplicated code on this file")
+      .setDebtSubCharacteristic(RulesDefinition.SubCharacteristics.LOGIC_CHANGEABILITY)
+      .setDebtRemediationFunction(rule.debtRemediationFunctions().linearWithOffset("10min", "10min"))
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "10min" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)